### PR TITLE
Add retries to inbound/outbound mail

### PR DESCRIPTION
**Pull request by dcramer**
This adds retries via `send_async`, and switches digests/invites to those.

@getsentry/infrastructure 

SENTRY-1AD

### CODE REVIEW

**Comment by mattrobenolt** [comment]
Looking at this quickly, the only hesitation is that `send_async()` wraps in `safe_execute`, so on failure to queue, we just move along. Is that acceptable here?

File: src/sentry/digests/backends/base.py [hunk 1]
```diff
@@ -153,7 +153,7 @@ def digest(self, key, minimum_delay=None):
             with timelines.digest('project:1') as records:
                 message = build_digest_email(records)
 
-            message.send()
+            message.send_async()
```
Comment (tkaemming) [inline src/sentry/digests/backends/base.py]: this is a docstring

Comment (dcramer) [inline src/sentry/digests/backends/base.py]: I realize that, but its what should be recommended.

**Comment by dcramer** [comment]
@mattrobenolt good point, I think we could kill send_async here, since the queue failing is critical. The problem is if things aren't transactional

**Pull request merged**